### PR TITLE
Auto reauthenticate account if expired

### PR DESCRIPTION
**Pull request by Trial97**
~blocked by https://github.com/PrismLauncher/PrismLauncher/pull/1771~

### CODE REVIEW

File: launcher/LaunchController.cpp
Comment (TheKodeToad) [inline launcher/LaunchController.cpp]: shouldn't this be removed as it's in #1771

Comment (Trial97) [reply launcher/LaunchController.cpp]: This is a blocked by the mentioned PR, please review it after that one was merged. Initially this was supposed to be one single PR but due to scrumplex request I removed the auto reauth part from that one( hope that one is merged faster that way). Because I thought that the auto reauth thing is still useful I open an additional PR with the revert of that last commit. So once that one is merged we can discuss here if we want that or not.

Comment (TheKodeToad) [reply launcher/LaunchController.cpp]: Oh. got it. I don't believe this has any reason to be based on the previous pr so I assumed it wasn't.

File: launcher/LaunchController.cpp
Comment (Ryex) [inline launcher/LaunchController.cpp]: does this need to be updated?

Comment (Trial97) [reply launcher/LaunchController.cpp]: In what way? Do you mean to bring it up to date with the latest develop?

Comment (Ryex) [reply launcher/LaunchController.cpp]: yes, the `MSALoginDialog::newAccount` call looks wrong for some reason.
also should we be changing the translation string considering mojang accounts don't really exist anymore (though it doesn't seem to actually be displayed anywhere)?

Comment (Trial97) [reply launcher/LaunchController.cpp]: That message is quickly overridden by another message (the subtask status message).
And no I did not have any issues with updating to latest develop.

**Review by Ryex — APPROVED** [review]

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**
